### ISSUE TITLE
htmlentities() expects parameter 1 to be string, array given

### ISSUE THREAD

**Issue by sjorsvanleeuwen** [LaravelCollective/html#433]
First things first, this is not a duplicate of #426 ;)

When creating a component i usually do stuff like
```php
array_merge_recursive(['class' => 'btn btn-default'], $attributes)
```

The idea is that the component always should have the classes btn and btn-default but the developer should be able to add more classes to it.
Works perfectly untill it needs to be rendered it ends up with the error specified in the title.

I did some digging and found a possible solution, in HtmlBuilder@attributeElement
```php
protected function attributeElement($key, $value)
    {
        // For numeric keys we will assume that the value is a boolean attribute
        // where the presence of the attribute represents a true value and the
        // absence represents a false value.
        // This will convert HTML attributes such as "required" to a correct
        // form instead of using incorrect numerics.
        if (is_numeric($key)) {
            return $value;
        }

        // Treat boolean attributes as HTML properties
        if (is_bool($value) && $key != 'value') {
            return $value ? $key : '';
        }
        
        // check for arrays and form them to a string too
        if (is_array($value)) {
            $value = implode(' ', $value);
        }

        if (! is_null($value)) {
            return $key . '="' . e($value) . '"';
        }
    }
```

More than happy to create a pull request of cource

**Comment by albertjt** [comment]
Are you sure that's the problem? I ended up in this issue since I got the same error, but after tweaking a bit I realize that my fault were because I forgot to initialize the parameter correctly on my page. Other than that it works perfectly. CMIIW

I did this that got that error
`Form::component('componentName', 'path-to-component', ['var1', 'var2' => null, 'var3' => []])`
`Form::componentName('var1', ['foo', 'bar']);`
should have been
`Form::componentName('var1', null, ['foo', 'bar'])`
